src/rest/server/handler.go

```diff
@@ -302,3 +310,64 @@ func hostMetadataHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/xrd+xml")
 	w.Write(data.Bytes())
 }
+
+func Authenticate(w http.ResponseWriter, r *http.Request) {
```

Comment:
Since JWT also involves authenticating via PAM initially, I would propose then moving basic auth into it's own file as well, and  have pamAuth.go as utility functions to do the user auth.

Reply:
See my latest commits